platform/mds/mds/src/main/java/org/motechproject/mds/service/impl/EntityServiceImpl.java

```diff
@@ -801,6 +802,23 @@ public void deleteEntity(Long entityId) {
         return entityDtos;
     }
 
+    @Override
+    @Transactional
+    public List<EntityDto> listEntitiesByBundle(String bundleSymbolicName) {
+        if (StringUtils.isBlank(bundleSymbolicName)) {
+            throw new MdsException("Bundle symbolic name cannot be empty or null");
+        }
+
+        List<EntityDto> entityDtos = new ArrayList<>();
+        for (Entity entity : allEntities.retrieveBySymbolicName(bundleSymbolicName)) {
```

Comment:
I meant that this could also go the repo class, since thats what we do in the other methods there.